src/antsibull_changelog/changes.py

```diff
@@ -416,7 +421,17 @@ def update_fragments(
     def sort(self) -> None:
         """
         Sort change metadata in place.
-        """
+        Optionally sorts by semantic version.
+        """
+        if self.config.changelog_semantic_versioning_sort:
+            self.data["releases"] = dict(
+                sorted(
+                    self.data["releases"].items(),
+                    key=lambda t: [int(v) for v in t[0].split(".")],
```

Comment:
That's definitely how you should NEVER sort versions :) Version parts can be alphanumerical, not just integers Think of `10.0.0rc1` (non-semver) or `1.0.0-rc1` (semver).

You need to use `self.version_constructor(version)` to convert a version string to a version object. These objects are comparable.